src/System.Linq/src/System/Linq/Lookup.cs

```diff
@@ -224,46 +273,59 @@ internal int InternalGetHashCode(TKey key)
         internal Grouping<TKey, TElement> GetGrouping(TKey key, bool create)
         {
             int hashCode = InternalGetHashCode(key);
-            for (Grouping<TKey, TElement> g = _groupings[hashCode % _groupings.Length]; g != null; g = g.hashNext)
-                if (g.hashCode == hashCode && _comparer.Equals(g.key, key)) return g;
+            for (Grouping<TKey, TElement> g = _groupings[hashCode % _groupings.Length]; g != null; g = g._hashNext)
+            {
+                if (g._hashCode == hashCode && _comparer.Equals(g._key, key))
+                {
+                    return g;
+                }
+            }
+
             if (create)
             {
-                if (_count == _groupings.Length) Resize();
+                if (_count == _groupings.Length)
+                {
+                    Resize();
+                }
+
                 int index = hashCode % _groupings.Length;
                 Grouping<TKey, TElement> g = new Grouping<TKey, TElement>();
-                g.key = key;
-                g.hashCode = hashCode;
-                g.elements = new TElement[1];
-                g.hashNext = _groupings[index];
+                g._key = key;
+                g._hashCode = hashCode;
+                g._elements = new TElement[1];
+                g._hashNext = _groupings[index];
                 _groupings[index] = g;
                 if (_lastGrouping == null)
                 {
-                    g.next = g;
+                    g._next = g;
                 }
                 else
                 {
-                    g.next = _lastGrouping.next;
-                    _lastGrouping.next = g;
+                    g._next = _lastGrouping._next;
+                    _lastGrouping._next = g;
                 }
+
                 _lastGrouping = g;
                 _count++;
                 return g;
             }
+
             return null;
         }
 
         private void Resize()
         {
-            int newSize = checked(_count * 2 + 1);
+            int newSize = checked((_count * 2) + 1);
```

Comment:
I personally find the previous more readable, because I'm led to wonder if the precedence is more complicated than it appears. I know that's subjective, and I won't argue about it if others disagree.